### PR TITLE
[16.0][ADD] base_wamas_ubl: Add new module to convert WAMAS files to UBL files and vice versa

### PR DESCRIPTION
**Pull request by tuantrantg**

### CODE REVIEW

File: base_wamas_ubl/lib/wamas/ubl_template/picking.xml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+				<cbc:TransportHandlingUnitTypeCode><t t-raw="line.package.IvTts_THM_ThmId"/></cbc:TransportHandlingUnitTypeCode>
+			</cac:TransportHandlingUnit>
+			<cac:GrossWeightMeasure>
+				<cbc:Measure><t t-raw="line.package.IvTek_GesGew"/></cbc:Measure>
+			</cac:GrossWeightMeasure>
+		</cac:Shipment>
+		<cbc:ID></cbc:ID>
+        <cbc:DeliveredQuantity t-att-unitCode="MAPPING['unitCode'][line.IvArt_HOSTUNITS_HostEinh] or MAPPING['unitCode'][line.IvArt_Bestand_Einheit]"><t t-raw="line.Mngs_Mng"/></cbc:DeliveredQuantity>
+		<cac:OrderLineReference>
+			<cbc:LineID></cbc:LineID>
```
Comment (nilshamerlinck) [inline base_wamas_ubl/lib/wamas/ubl_template/picking.xml]: we can probably use `<t t-raw="record_index"/>` here as well to get a valid value

File: base_wamas_ubl/lib/wamas/ubl2wamas.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    from .wamas_grammar import ausk, ausp, weak, weap  # noqa: F401
+except ImportError:
+    from utils import *  # noqa: F403
+    from wamas_grammar import ausk, ausp, weak, weap  # noqa: F401
+
+
+LST_FIELD_UNIT_CODE = ["HostEinheit"]
+
+
+MAPPING_UBL_TO_WAMAS = {
```
Comment (nilshamerlinck) [inline base_wamas_ubl/lib/wamas/ubl2wamas.py]: hmm can't build it from the existing mapping?

File: base_wamas_ubl/lib/wamas/ubl2wamas.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+            "LTR": "LITRE",  # litre
+            "XBO": "PET",  # glass bottle
+            "X4B": "TETRA",  # tetra pack
+            "": False,  # undefined,
+        }
+    )
+}
+
+
+def _set_string(val, length, dp):
```
Comment (nilshamerlinck) [inline base_wamas_ubl/lib/wamas/ubl2wamas.py]: could you add a `test_ubl2wamas.py` that tests for expected values of these helpers?

**Review by nilshamerlinck — CHANGES_REQUESTED** [review]

**Comment by tuantrantg** [comment]
hi @nilshamerlinck please review your requests again, I've updated the PR

**Pull request closed without merging**